apps/documents/readers.py

```
@@ -68,10 +69,16 @@ def markitdown_read(file_obj) -> Document:
 def plaintext_reader(file_obj) -> Document:
+    content = file_obj.read()
     try:
-        return Document(parts=[DocumentPart(content=file_obj.read().decode())])
-    except UnicodeDecodeError as e:
+        # UTF-8 decode
+        content = content.decode()
+    except UnicodeDecodeError:
+        # Try to detect encoding
+        content = UnicodeDammit(content).unicode_markup
+    except Exception as e:
         raise FileReadException("Unable to decode file contents to text") from e
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)